core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -205,8 +206,12 @@ protected void add(ManifestFile manifest) {
     // the manifest must be rewritten with this update's snapshot ID
     try (ManifestReader reader = ManifestReader.read(
         ops.io().newInputFile(manifest.path()), ops.current().specsById())) {
-      appendManifests.add(ManifestWriter.copyAppendManifest(
-          reader, manifestPath(manifestCount.getAndIncrement()), snapshotId(), appendedManifestsSummary));
+      ManifestFile manifestFile = ManifestWriter.copyAppendManifest(
+          reader, manifestPath(manifestCount.getAndIncrement()), snapshotId(), appendedManifestsSummary);
+      appendManifests.add(manifestFile);
+      // keep reference to the latest appended manifest, so that we can avoid merging bins which are
+      // have not crossed the limit of minManifestsCountToMerge
+      cachedNewAppenedManifest = manifestFile;
```

Comment:
I would prefer a more specific name for this, like `lastAppendedManifest`. It isn't clear from the name that this is intended to be replaced, so I initially thought it was a bug before seeing how it was used.

Reply:
Also, this has a typo: `Appened` should be `Appended`.

Reply:
used `firstAppendedManifest`.